sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/MessageFlux.java

```diff
@@ -0,0 +1,1045 @@
+         *
+         * @param parent the parent message-flux.
+         * @param messageSubscriber the message-flux's downstream subscriber to notify the events.
+         * @param prefetch the number of messages that the operator should prefetch from the messaging entity
+         *                 (for a less chatty network and faster message processing on the client).
+         * @param creditFlowMode the mode indicating how to compute the credit and when to send it to the broker.
+         * @param retryPolicy the retry policy to use to recover from receiver termination.
+         */
+        RecoverableReactorReceiver(MessageFlux parent, CoreSubscriber<? super Message> messageSubscriber, int prefetch,
+            CreditFlowMode creditFlowMode, AmqpRetryPolicy retryPolicy) {
+            this.parent = parent;
+            this.messageSubscriber = messageSubscriber;
+            this.prefetch = prefetch;
+            this.creditFlowMode = creditFlowMode;
+            this.retryPolicy = retryPolicy;
+            this.logger = parent.logger;
+        }
+
+        /**
+         * Invoked by the upstream in response to message-flux subscribing to it.
+         *
+         * @param s the subscription handle for requesting receivers from the upstream or terminating upstream
+         *         through cancellation when it is no longer needed.
+         */
+        @Override
+        public void onSubscribe(Subscription s) {
+            if (Operators.validate(upstream, s)) {
+                upstream = s;
+                // Invoke the downstream 'onSubscribe' with the subscription handle enabling downstream to
+                // request messages or cancellation.
+                messageSubscriber.onSubscribe(this);
+                // Request the first receiver from the upstream.
+                s.request(1);
+            }
+        }
+
+        /**
+         * Invoked by the upstream to deliver new receiver.
+         *
+         * @param receiver the new receiver.
+         */
+        @Override
+        public void onNext(AmqpReceiveLink receiver) {
+            if (done) {
+                receiver.closeAsync().subscribe();
+                Operators.onNextDropped(receiver, messageSubscriber.currentContext());
+                return;
+            }
+
+            // Create a new mediator to channel communication between the new receiver and the recoverable-receiver.
+            final ReactorReceiverMediator mediator = new ReactorReceiverMediator(this, receiver, prefetch, creditFlowMode, logger);
+
+            // Request MediatorHolder to set the new mediator as the current (for the drain-loop to pick)
+            if (mediatorHolder.trySet(mediator)) {
+                // the MediatorHolder accepted the mediator. Notify the mediator that the recoverable-receiver
+                // (a.k.a parent) is ready to use the mediator; in-response, the mediator notifies its readiness
+                // by invoking 'onMediatorReady'.
```

Comment:
Mismatched method name with comment.

Reply:
Ah, that was a miss, I’ll make it private.

Reply:
Let me update the doc to make it more redable.